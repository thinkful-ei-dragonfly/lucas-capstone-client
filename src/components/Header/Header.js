

import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import IdleService from '../../services/idle-service'

export default class Header extends React.Component {
  handleLogoutClick = () => {
    TokenService.clearAuthToken()
    /* when logging out, clear the callbacks to the refresh api and idle auto logout */
    TokenService.clearCallbackBeforeExpiry()
    IdleService.unRegisterIdleResets()
  }

  renderLogoutLink() {
    return (
      <div className='Header__logged-in'>
        <Link
          onClick={this.handleLogoutClick}
          to='/'>
          Logout
        </Link>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <div className='Header__not-logged-in'>
        <Link
          to='/login'>
          Log in
        </Link>
        {' '}
        <Link
          to='/register'>
          Register
        </Link>
      </div>
    )
  }

  render() {
    const newVariable = 'hello'
    return (
      <header className="Header">
        <h1 className="header-h1">
          <Link to='/'>Colecci√≥n</Link>
        </h1>
        {TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()}
      </header>
    )
  }
}
