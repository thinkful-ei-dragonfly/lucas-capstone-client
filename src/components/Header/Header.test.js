import React from 'react'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'

import Header from './Header'
configure({ adapter: new Adapter() });
describe('Header component', () => {
  const props = {
    something: 'something'
  }

  it('renders a Header by default', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders a h1 with app name', () => {
    const h1 = shallow(<Header />)
      .find('.header-h1')
    expect(toJson(h1)).toMatchSnapshot()
  })
})
