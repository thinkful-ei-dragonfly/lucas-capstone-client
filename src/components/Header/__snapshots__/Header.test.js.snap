// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a Header by default 1`] = `
<header
  className="Header"
>
  <h1
    aria-label="App Name"
    className="header-h1"
    role="banner"
  >
    <Link
      to="/"
    >
      Colección
    </Link>
  </h1>
  <div
    aria-label="Header Navigation"
    className="logged-out"
    role="navigation"
  >
    <Link
      to="/about"
    >
      About
    </Link>
    <Link
      to="/login"
    >
      Log in
    </Link>
    <Link
      to="/register"
    >
      Register
    </Link>
  </div>
</header>
`;

exports[`Header component renders a h1 with app name 1`] = `
<h1
  aria-label="App Name"
  className="header-h1"
  role="banner"
>
  <Link
    to="/"
  >
    Colección
  </Link>
</h1>
`;
